lib/core/src/support/Utils.ts

```diff
@@ -21,17 +21,21 @@ export class Utils {
 
     const edge: boolean = chrome && (navigator.userAgent.indexOf('Edg') !== -1)
 
-    const extensions: boolean = typeof (window as any).InstallTrigger !== 'undefined'
+    const firefox: boolean = typeof (window as any).InstallTrigger !== 'undefined'
       || (!(document as any).documentNode && (window as any).StyleMedia)
 
     const opera: boolean = (!!(window as any).opr && !!(window as any).addons)
       || !!(window as any).opera || navigator.userAgent.indexOf(' OPR/') >= 0
 
-    // Currently we will not return `Browser.OPERA` but fall back to `.CHROME`
-    return extensions ? Browser.EXTENSIONS
+    const safari: boolean = /constructor/i.test((window as any).HTMLElement)
```

Comment:
It'd be nice to include a link from which this detector was taken. If something breaks the original link might have a fix.

Reply:
I borrowed it from the stackoverflow resource from the comment above 🙂 
```
// Methods are borrowed from the following SO resource which is regularly updated:
// https://stackoverflow.com/questions/9847580/how-to-detect-safari-chrome-ie-firefox-and-opera-browser#9851769
```